### PR TITLE
Update to nightly rustc to 2023-04-19

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

I'm impatient. ;)

and our nightly is too old

#### Summary of changes

Update nightly.

### CODE REVIEW

**Comment by yihau** [comment]
just updated docker images and retried the pipeline! let's see how it goes 👻

File: scripts/coverage.sh [hunk 1]
```diff
@@ -42,6 +42,7 @@ coverageFlags+=("-Ccodegen-units=1")     # Disable code generation parallelism w
 coverageFlags+=("-Cinline-threshold=0")  # Disable inlining, which complicates control flow.
 coverageFlags+=("-Copt-level=0")
 coverageFlags+=("-Coverflow-checks=off") # Disable overflow checks, which create unnecessary branches.
+coverageFlags+=("-Cdebug-assertions=off") # aaaa
```
Comment (ryoqun) [inline scripts/coverage.sh:45]: here

File: gossip/src/cluster_info.rs [hunk 2]
```diff
@@ -273,7 +273,7 @@ pub fn make_accounts_hashes_message(
 pub(crate) type Ping = ping_pong::Ping<[u8; GOSSIP_PING_TOKEN_SIZE]>;
 
 // TODO These messages should go through the gpu pipeline for spam filtering
-#[frozen_abi(digest = "FsZnSeTYNH7F51AxTaKUixXxjT6if2ThmPN1mhDWtXZM")]
+#[frozen_abi(digest = "Ctxue3UVFXXqnHoMVAPmfBoCy3Cyg7gNCYBY7Cg9P3so")]
```
Comment (ryoqun) [inline gossip/src/cluster_info.rs:276]: this is expected according to this: https://github.com/rust-lang/rust/pull/104265

the actual abi change is below:

```patch
diff --git a/tmp/abi8/cluster_info__Protocol_frozen_abi__test_abi_digest_FsZnSeTYNH7F51AxTaKUixXxjT6if2ThmPN1mhDWtXZM b/tmp/abi8/cluster_info__Protocol_frozen_abi__test_abi_digest_Ctxue3UVFXXqnHoMVAPmfBoCy3Cyg7gNCYBY7Cg9P3so
index a5d2d7bc63..59986592ba 100644
--- a/tmp/abi8/cluster_info__Protocol_frozen_abi__test_abi_digest_FsZnSeTYNH7F51AxTaKUixXxjT6if2ThmPN1mhDWtXZM
+++ b/tmp/abi8/cluster_info__Protocol_frozen_abi__test_abi_digest_Ctxue3UVFXXqnHoMVAPmfBoCy3Cyg7gNCYBY7Cg9P3so
@@ -232,25 +232,25 @@ enum Protocol (variants = 6)
                                                 primitive u8
                                             element u8
                                                 primitive u8
-                                field gossip: std::net::socket_addr::SocketAddr
+                                field gossip: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field tvu: std::net::socket_addr::SocketAddr
+                                field tvu: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field tvu_forwards: std::net::socket_addr::SocketAddr
+                                field tvu_forwards: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field repair: std::net::socket_addr::SocketAddr
+                                field repair: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field tpu: std::net::socket_addr::SocketAddr
+                                field tpu: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field tpu_forwards: std::net::socket_addr::SocketAddr
+                                field tpu_forwards: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field tpu_vote: std::net::socket_addr::SocketAddr
+                                field tpu_vote: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field rpc: std::net::socket_addr::SocketAddr
+                                field rpc: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                                field rpc_pubsub: core::net::socket_addr::SocketAddr
                                     primitive str
-                                field serve_repair: std::net::socket_addr::SocketAddr
+                                field serve_repair: core::net::socket_addr::SocketAddr
                                     primitive str
                                 field wallclock: u64
                                     primitive u64
@@ -1789,7 +1789,7 @@ enum Protocol (variants = 6)
                                             tuple (elements = 1)
                                                 element u8
                                                     primitive u8
-                                        element std::net::ip_addr::IpAddr
+                                        element core::net::ip_addr::IpAddr
                                             primitive str
                                 field sockets: __SerializeWith
                                     tuple (elements = 1)
@@ -2081,25 +2081,25 @@ enum Protocol (variants = 6)
                                                         primitive u8
                                                     element u8
                                                         primitive u8
-                                        field gossip: std::net::socket_addr::SocketAddr
+                                        field gossip: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tvu: std::net::socket_addr::SocketAddr
+                                        field tvu: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tvu_forwards: std::net::socket_addr::SocketAddr
+                                        field tvu_forwards: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field repair: std::net::socket_addr::SocketAddr
+                                        field repair: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tpu: std::net::socket_addr::SocketAddr
+                                        field tpu: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tpu_forwards: std::net::socket_addr::SocketAddr
+                                        field tpu_forwards: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tpu_vote: std::net::socket_addr::SocketAddr
+                                        field tpu_vote: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field rpc: std::net::socket_addr::SocketAddr
+                                        field rpc: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field rpc_pubsub: std::net::socket_addr::SocketAddr
+                                        field rpc_pubsub: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field serve_repair: std::net::socket_addr::SocketAddr
+                                        field serve_repair: core::net::socket_addr::SocketAddr
                                             primitive str
                                         field wallclock: u64
                                             primitive u64
@@ -3638,7 +3638,7 @@ enum Protocol (variants = 6)
                                                     tuple (elements = 1)
                                                         element u8
                                                             primitive u8
-                                                element std::net::ip_addr::IpAddr
+                                                element core::net::ip_addr::IpAddr
                                                     primitive str
                                         field sockets: __SerializeWith
                                             tuple (elements = 1)
@@ -3930,25 +3930,25 @@ enum Protocol (variants = 6)
                                                         primitive u8
                                                     element u8
                                                         primitive u8
-                                        field gossip: std::net::socket_addr::SocketAddr
+                                        field gossip: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tvu: std::net::socket_addr::SocketAddr
+                                        field tvu: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tvu_forwards: std::net::socket_addr::SocketAddr
+                                        field tvu_forwards: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field repair: std::net::socket_addr::SocketAddr
+                                        field repair: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tpu: std::net::socket_addr::SocketAddr
+                                        field tpu: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tpu_forwards: std::net::socket_addr::SocketAddr
+                                        field tpu_forwards: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field tpu_vote: std::net::socket_addr::SocketAddr
+                                        field tpu_vote: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field rpc: std::net::socket_addr::SocketAddr
+                                        field rpc: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field rpc_pubsub: std::net::socket_addr::SocketAddr
+                                        field rpc_pubsub: core::net::socket_addr::SocketAddr
                                             primitive str
-                                        field serve_repair: std::net::socket_addr::SocketAddr
+                                        field serve_repair: core::net::socket_addr::SocketAddr
                                             primitive str
                                         field wallclock: u64
                                             primitive u64
@@ -5487,7 +5487,7 @@ enum Protocol (variants = 6)
                                                     tuple (elements = 1)
                                                         element u8
                                                             primitive u8
-                                                element std::net::ip_addr::IpAddr
+                                                element core::net::ip_addr::IpAddr
                                                     primitive str
                                         field sockets: __SerializeWith
                                             tuple (elements = 1)
```

File: core/src/serve_repair.rs [hunk 3]
```diff
@@ -226,7 +226,7 @@ pub(crate) type Ping = ping_pong::Ping<[u8; REPAIR_PING_TOKEN_SIZE]>;
 
 /// Window protocol messages
 #[derive(Debug, AbiEnumVisitor, AbiExample, Deserialize, Serialize)]
-#[frozen_abi(digest = "3bgE3sYHRqetvpo4fcDL6PTV3z2LMAtY6H8BoLFSjCwf")]
+#[frozen_abi(digest = "6VyBwHjkAMXAN97fdhQgFv6VdPEnfJo9LdUAd2SFtwF3")]
```
Comment (ryoqun) [inline core/src/serve_repair.rs:229]: same as https://github.com/solana-labs/solana/pull/31381/files#r1184956432

```patch
diff --git a/tmp/abi8/serve_repair__RepairProtocol_frozen_abi__test_abi_digest_3bgE3sYHRqetvpo4fcDL6PTV3z2LMAtY6H8BoLFSjCwf b/tmp/abi8/serve_repair__RepairProtocol_frozen_abi__test_abi_digest_6VyBwHjkAMXAN97fdhQgFv6VdPEnfJo9LdUAd2SFtwF3
index ea90899190..10bf49ac51 100644
--- a/tmp/abi8/serve_repair__RepairProtocol_frozen_abi__test_abi_digest_3bgE3sYHRqetvpo4fcDL6PTV3z2LMAtY6H8BoLFSjCwf
+++ b/tmp/abi8/serve_repair__RepairProtocol_frozen_abi__test_abi_digest_6VyBwHjkAMXAN97fdhQgFv6VdPEnfJo9LdUAd2SFtwF3
@@ -69,25 +69,25 @@ enum RepairProtocol (variants = 12)
                                 primitive u8
                             element u8
                                 primitive u8
-                field gossip: std::net::socket_addr::SocketAddr
+                field gossip: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu: std::net::socket_addr::SocketAddr
+                field tvu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu_forwards: std::net::socket_addr::SocketAddr
+                field tvu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field repair: std::net::socket_addr::SocketAddr
+                field repair: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu: std::net::socket_addr::SocketAddr
+                field tpu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_forwards: std::net::socket_addr::SocketAddr
+                field tpu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_vote: std::net::socket_addr::SocketAddr
+                field tpu_vote: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc: std::net::socket_addr::SocketAddr
+                field rpc: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                field rpc_pubsub: core::net::socket_addr::SocketAddr
                     primitive str
-                field serve_repair: std::net::socket_addr::SocketAddr
+                field serve_repair: core::net::socket_addr::SocketAddr
                     primitive str
                 field wallclock: u64
                     primitive u64
@@ -167,25 +167,25 @@ enum RepairProtocol (variants = 12)
                                 primitive u8
                             element u8
                                 primitive u8
-                field gossip: std::net::socket_addr::SocketAddr
+                field gossip: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu: std::net::socket_addr::SocketAddr
+                field tvu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu_forwards: std::net::socket_addr::SocketAddr
+                field tvu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field repair: std::net::socket_addr::SocketAddr
+                field repair: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu: std::net::socket_addr::SocketAddr
+                field tpu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_forwards: std::net::socket_addr::SocketAddr
+                field tpu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_vote: std::net::socket_addr::SocketAddr
+                field tpu_vote: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc: std::net::socket_addr::SocketAddr
+                field rpc: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                field rpc_pubsub: core::net::socket_addr::SocketAddr
                     primitive str
-                field serve_repair: std::net::socket_addr::SocketAddr
+                field serve_repair: core::net::socket_addr::SocketAddr
                     primitive str
                 field wallclock: u64
                     primitive u64
@@ -265,25 +265,25 @@ enum RepairProtocol (variants = 12)
                                 primitive u8
                             element u8
                                 primitive u8
-                field gossip: std::net::socket_addr::SocketAddr
+                field gossip: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu: std::net::socket_addr::SocketAddr
+                field tvu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu_forwards: std::net::socket_addr::SocketAddr
+                field tvu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field repair: std::net::socket_addr::SocketAddr
+                field repair: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu: std::net::socket_addr::SocketAddr
+                field tpu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_forwards: std::net::socket_addr::SocketAddr
+                field tpu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_vote: std::net::socket_addr::SocketAddr
+                field tpu_vote: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc: std::net::socket_addr::SocketAddr
+                field rpc: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                field rpc_pubsub: core::net::socket_addr::SocketAddr
                     primitive str
-                field serve_repair: std::net::socket_addr::SocketAddr
+                field serve_repair: core::net::socket_addr::SocketAddr
                     primitive str
                 field wallclock: u64
                     primitive u64
@@ -361,25 +361,25 @@ enum RepairProtocol (variants = 12)
                                 primitive u8
                             element u8
                                 primitive u8
-                field gossip: std::net::socket_addr::SocketAddr
+                field gossip: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu: std::net::socket_addr::SocketAddr
+                field tvu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu_forwards: std::net::socket_addr::SocketAddr
+                field tvu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field repair: std::net::socket_addr::SocketAddr
+                field repair: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu: std::net::socket_addr::SocketAddr
+                field tpu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_forwards: std::net::socket_addr::SocketAddr
+                field tpu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_vote: std::net::socket_addr::SocketAddr
+                field tpu_vote: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc: std::net::socket_addr::SocketAddr
+                field rpc: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                field rpc_pubsub: core::net::socket_addr::SocketAddr
                     primitive str
-                field serve_repair: std::net::socket_addr::SocketAddr
+                field serve_repair: core::net::socket_addr::SocketAddr
                     primitive str
                 field wallclock: u64
                     primitive u64
@@ -461,25 +461,25 @@ enum RepairProtocol (variants = 12)
                                 primitive u8
                             element u8
                                 primitive u8
-                field gossip: std::net::socket_addr::SocketAddr
+                field gossip: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu: std::net::socket_addr::SocketAddr
+                field tvu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu_forwards: std::net::socket_addr::SocketAddr
+                field tvu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field repair: std::net::socket_addr::SocketAddr
+                field repair: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu: std::net::socket_addr::SocketAddr
+                field tpu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_forwards: std::net::socket_addr::SocketAddr
+                field tpu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_vote: std::net::socket_addr::SocketAddr
+                field tpu_vote: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc: std::net::socket_addr::SocketAddr
+                field rpc: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                field rpc_pubsub: core::net::socket_addr::SocketAddr
                     primitive str
-                field serve_repair: std::net::socket_addr::SocketAddr
+                field serve_repair: core::net::socket_addr::SocketAddr
                     primitive str
                 field wallclock: u64
                     primitive u64
@@ -561,25 +561,25 @@ enum RepairProtocol (variants = 12)
                                 primitive u8
                             element u8
                                 primitive u8
-                field gossip: std::net::socket_addr::SocketAddr
+                field gossip: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu: std::net::socket_addr::SocketAddr
+                field tvu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu_forwards: std::net::socket_addr::SocketAddr
+                field tvu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field repair: std::net::socket_addr::SocketAddr
+                field repair: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu: std::net::socket_addr::SocketAddr
+                field tpu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_forwards: std::net::socket_addr::SocketAddr
+                field tpu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_vote: std::net::socket_addr::SocketAddr
+                field tpu_vote: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc: std::net::socket_addr::SocketAddr
+                field rpc: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                field rpc_pubsub: core::net::socket_addr::SocketAddr
                     primitive str
-                field serve_repair: std::net::socket_addr::SocketAddr
+                field serve_repair: core::net::socket_addr::SocketAddr
                     primitive str
                 field wallclock: u64
                     primitive u64
@@ -659,25 +659,25 @@ enum RepairProtocol (variants = 12)
                                 primitive u8
                             element u8
                                 primitive u8
-                field gossip: std::net::socket_addr::SocketAddr
+                field gossip: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu: std::net::socket_addr::SocketAddr
+                field tvu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tvu_forwards: std::net::socket_addr::SocketAddr
+                field tvu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field repair: std::net::socket_addr::SocketAddr
+                field repair: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu: std::net::socket_addr::SocketAddr
+                field tpu: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_forwards: std::net::socket_addr::SocketAddr
+                field tpu_forwards: core::net::socket_addr::SocketAddr
                     primitive str
-                field tpu_vote: std::net::socket_addr::SocketAddr
+                field tpu_vote: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc: std::net::socket_addr::SocketAddr
+                field rpc: core::net::socket_addr::SocketAddr
                     primitive str
-                field rpc_pubsub: std::net::socket_addr::SocketAddr
+                field rpc_pubsub: core::net::socket_addr::SocketAddr
                     primitive str
-                field serve_repair: std::net::socket_addr::SocketAddr
+                field serve_repair: core::net::socket_addr::SocketAddr
                     primitive str
                 field wallclock: u64
                     primitive u64

```

File: ci/rust-version.sh [hunk 4]
```diff
@@ -29,7 +29,7 @@ fi
 if [[ -n $RUST_NIGHTLY_VERSION ]]; then
   nightly_version="$RUST_NIGHTLY_VERSION"
 else
-  nightly_version=2023-01-22
+  nightly_version=2023-04-19
```
[the remark this answers is not in the document]
Comment (ryoqun) [reply ci/rust-version.sh:32]: hm? why bumping the nightly version is needed?

File: programs/bpf_loader/src/serialization.rs [hunk 5]
```diff
@@ -994,58 +995,94 @@ mod tests {
     }
 
     // the old bpf_loader in-program deserializer bpf_loader::id()
-    #[allow(clippy::type_complexity)]
+    #[deny(unsafe_op_in_unsafe_fn)]
```
Comment (ryoqun) [inline programs/bpf_loader/src/serialization.rs:998]: personally, i want this to be enabled at tree-wide. but this is different story.

File: programs/bpf_loader/src/serialization.rs [hunk 6]
```diff
@@ -994,58 +995,94 @@ mod tests {
...
+            }
+
+            // rustc inserts debug_assert! for misaligned pointer dereferences when
+            // deserializing, starting from [1]. so, use std::mem::transmute as the last resort
+            // while preventing clippy from complaining to suggest not to use it.
+            // [1]: https://github.com/rust-lang/rust/commit/22a7a19f9333bc1fcba97ce444a3515cb5fb33e6
+            // as for the ub nature of the misaligned pointer dereference, this is
+            // acceptable in this code, given that this is cfg(test) and it's cared only with
+            // x86-64 and the target only incurs some performance penalty, not like segfaults
+            // in other targets.
```
Comment (ryoqun) [inline programs/bpf_loader/src/serialization.rs:1026]: @dmakarov this is fyi. I think you'll need to update the original code at `sdk/program/entrypoint{,_deprecated}.rs` when updating platform-tool to be based on rust toolchain 1.70+.

I already chatted with @Lichtso and @alessandrod. hope this comment does make sense for the uninformed. :)

File: ci/test-checks.sh [hunk 7]
```diff
@@ -75,7 +75,7 @@ fi
 
  _ ci/order-crates-for-publishing.py
 
-nightly_clippy_allows=()
+nightly_clippy_allows=(--allow=clippy::redundant_clone --allow clippy::match-result-ok)
```
[the remark this answers is not in the document]
Comment (ryoqun) [reply ci/test-checks.sh:78]: to be honest, i haven't looked into it at all and i'm just too tired at this moment. ;)

(rant time: this rust bump was simply cursed: sscache upstream debug and fix (thanks for the hard work by @yihau), grcov racing-condition bug (which turned out false-positive), the new misaligned debug_assert (see below diff), odd sigill in coverage update (ref: #31099); i just wanted `Arc::into_inner()`, which was merged a day after the nightly before #31487...)

File: programs/bpf_loader/src/serialization.rs
Comment (ryoqun) [inline programs/bpf_loader/src/serialization.rs:1]: @Lichtso could you review changes in this file? i think it's the only review-worth diff in this pr to merge. :)

File: ci/test-checks.sh (hunk 7 above)
[the remark this answers is not in the document]
Comment (ryoqun) [reply ci/test-checks.sh]: the `checks` step just passed with ef8bb88a7d4. :)

**Review by Lichtso — APPROVED** [review]

File: ci/test-checks.sh [hunk 8]
```diff
@@ -75,7 +75,7 @@ fi
 
  _ ci/order-crates-for-publishing.py
 
-nightly_clippy_allows=()
+nightly_clippy_allows=(--allow=clippy::redundant_clone)
```
Comment (CriesofCarrots) [inline ci/test-checks.sh:78]: Why did you add this @ryoqun ?

Comment (ryoqun) [reply ci/test-checks.sh:78]: @CriesofCarrots in short, I was exhausted for bumping nightly this time, which took months. (details: https://github.com/solana-labs/solana/pull/31381#discussion_r1186698539)

just saw #31690. sorry for trouble because of inter-branch clippy inconsistency. to be honest, i didn't anticipated 
this.

after initial triage from @brooksprumo, i sensed fixing this isn't straigtforward: https://github.com/solana-labs/solana/pull/31381#discussion_r1186711957

so, i opted to merge this pr in as-is. (this urgency is coming from the fact my upcoming pr will be blocked on rustc 1.70+)

seems @apfitzge is starting to tackle on this: #31685.

initially, i thought this exception isn't end of world thing, but seems to cause actual annoyance. I'll work on this.

so, put differently, crude way to solicit collaboration. ;)
